students/rmart300/trigram/trigram.py

```diff
@@ -1,45 +1,60 @@
+import sys, random, re
 
 trigram_dict={}
 
 def read_book(book_file):
-
-    book = open(book_file, 'r'):
-    digram = ''
-    for line in book:
-        line.strip()
-        words = line.split()
-        for i in range(len(words)):
-            if i > 1:
-                digram = f"{words[i-1]} {words[i]}"
-            if i < len(words)-1:
-                if digram in triagram_dict:
-                    word_list = trigram_dict[digram]
-                    word_list.append(words[i+1])
-                    trigram_dict[digram] = wordlist
-                else:
-                    trigram_dict[digram] = []
+    """read the book and make trigram dict"""
+    with open(book_file, 'r') as book:
+        digram = ''
+        for line in book:
+            #need to remove punctuation from words
+            punc = "\W"
+            line = re.sub(punc, ' ', line.strip().lower())
+            words = line.split()
+
+            for i in range(len(words)):
+                if i > 1:
+                    digram = f"{words[i-1]} {words[i]}"
+                if i < len(words)-1:
+                    if digram in trigram_dict:
+                        trigram_dict[digram].append(words[i+1])
+                    else:
+                        trigram_dict[digram] = [words[i+1]]
 
 def write_book():
+    """write book in trigram syntax"""
     trigram_usage_dict = {}
     for digram in trigram_dict:
         trigram_usage_dict[digram] = 0
```

Comment:
I think something like Counter would be cool to use here:
https://docs.python.org/2/library/collections.html#collections.Counter

here is a demonstration:
```
c = Counter(dict.fromkeys(trigram_dict.keys(), 0))

# to increment
c[digram] += 1

```

what you have is totally valid, just exploring other features of dict 🙂